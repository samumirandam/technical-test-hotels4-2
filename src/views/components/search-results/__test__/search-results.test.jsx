/* eslint-disable import/no-unresolved */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render, screen } from '@utils/test-utils';

import SearchResults from '../index';

const defaultProps = {};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<SearchResults {...setupStore} />, {});
};

describe('Test for SearchResults component', () => {
  test('Should render without errors', () => {
    setup();
    expect(screen.getByText('Hoteles en:')).toBeTruthy();
  });

  test('Should renders the same component', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test('Should render with data', () => {
    const props = {
      destination: 'destinationTest',
      adults: '1',
      checkIn: '11-11-2022',
      checkOut: '15-11-2022',
      pageSize: '1',
      totalResults: '1',
    };
    setup(props);
    expect(screen.getByText(`Hoteles en: ${props.destination}`)).toBeTruthy();
    expect(screen.getByText(`Para ${props.adults} personas`)).toBeTruthy();
  });
});
