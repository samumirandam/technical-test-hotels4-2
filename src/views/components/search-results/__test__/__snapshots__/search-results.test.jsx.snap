// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for SearchResults component Should renders the same component 1`] = `
<div>
  <div
    class="SearchResults"
  >
    <h1
      class="SearchResults__title"
    >
      Hotels in:
       
       
    </h1>
    <p />
  </div>
</div>
`;
